core/service/transaction.ts

```diff
@@ -1,50 +1,42 @@
 import crypto from 'crypto';
 import cryptoBrowserify from 'crypto-browserify';
 
-import { IAsset, Transaction, TransactionType, TransactionModel } from 'shared/model/transaction';
+import { IAsset, Transaction, TransactionType } from 'shared/model/transaction';
 import { IFunctionResponse, ITableObject } from 'core/util/common';
 import ResponseEntity from 'shared/model/response';
 import { ed, IKeyPair } from 'shared/util/ed';
 import { Account, Address } from 'shared/model/account';
 import config from 'shared/util/config';
-import AccountRepo from '../repository/account';
-import TransactionRepo from '../repository/transaction';
-import TransactionPool from './transactionPool';
-import TransactionQueue from './transactionQueue';
+import AccountRepo from 'core/repository/account';
+import TransactionRepo from 'core/repository/transaction';
+import TransactionPool from 'core/service/transactionPool';
+import TransactionQueue from 'core/service/transactionQueue';
 import { transactionSortFunc, getTransactionServiceByType, TRANSACTION_BUFFER_SIZE } from 'core/util/transaction';
 import BUFFER from 'core/util/buffer';
 import { SALT_LENGTH } from 'core/util/const';
 
-export interface ITransactionService<T extends IAsset> {
-    getBytes(trs: TransactionModel<T>): Buffer;
+export interface IAssetService<T extends IAsset> {
+    getBytes(trs: Transaction<T>): Buffer;
 
-    create(trs: TransactionModel<T>, data?: T): void;
+    create(trs: Transaction<T>, data?: T): void;
 
-    verify(trs: TransactionModel<T>, sender: Account): ResponseEntity<void>;
+    validate(trs: Transaction<T>, sender: Account): ResponseEntity<void>;
     verifyUnconfirmed(trs: Transaction<T>, sender: Account): ResponseEntity<void>;
 
     applyUnconfirmed(trs: Transaction<T>, sender: Account): ResponseEntity<void>;
     undoUnconfirmed(trs: Transaction<T>, sender: Account): ResponseEntity<void>;
 
     calculateUndoUnconfirmed(trs: Transaction<T>, sender: Account): void;
-
-    calculateFee(trs: TransactionModel<IAsset>, sender: Account): number;
-
-    apply(trs: Transaction<T>, sender: Account): Promise<ResponseEntity<void>>;
-    undo(trs: Transaction<T>, sender: Account): Promise<ResponseEntity<void>>;
+    calculateFee(trs: Transaction<IAsset>, sender: Account): number;
 }
 
-export interface ITransactionDispatcher<T extends IAsset> {
-    // list(): any; // to repo
-    // getById(): any; // to repo
-
-    // getVotesById(): any; // ?
+export interface ITransactionService<T extends IAsset> {
 
     checkSenderTransactions(
         senderAddress: Address, verifiedTransactions: Set<string>, accountsMap: { [address: string]: Account }
     ): Promise<void>;
 
-    verify(trs: TransactionModel<T>, sender: Account): ResponseEntity<void>;
+    validate(trs: Transaction<T>, sender: Account): ResponseEntity<void>;
```

Comment:
sender is not needed in the validate method

Reply:
future refactor